src/main/kotlin/com/exactpro/th2/conn/amqp/client/AmqpClient.kt

```diff
@@ -55,26 +56,31 @@ class AmqpClient(config: Config, val errorReporter: (Throwable) -> Unit) : IClie
             receiveDestination = context.lookup("receiveQueue") as Destination
         }
 
+        sessionAlias = properties[sessionAliasPropertyName] as String
         jmsContext.start()
         LOGGER.info("Connected to amqp broker successfully")
 
         consumer = jmsContext.createConsumer(receiveDestination) as JmsConsumer
         LOGGER.info("Queue consumer created to read data form the Queue:  {}", receiveDestination)
 
         producer = jmsContext.createProducer() as JmsProducer
+
+        producer.setProperty(sessionAliasPropertyName, sessionAlias);
     }
 
     override fun setMessageListener(callback: (ByteArray) -> Unit) {
         LOGGER.debug("Set an asynchronous queue listener")
         consumer.messageListener = MessageListener { message: Message ->
-            LOGGER.debug("Message received from the Queue:  {}", receiveDestination)
-            message.runCatching(Companion::toBytes).onSuccess(callback).onFailure {
-                LOGGER.error(it) {"Error while processing incoming message"}
-                errorReporter(it)
-            }
-            message.runCatching(Message::acknowledge).onFailure {
-                LOGGER.error(it) {"Error while acknowledging received message"}
-                errorReporter(it)
+            if (sessionAlias == message.getStringProperty("sessionAlias")) {
```

Comment:
Remote server doesn't know about th2 and its session alias then incoming AMQP messages don't contain session_alias property.
Current conn creates distinct AMQPClient for each session_alias then all consumed messages can be marked by session_alias passed to a constructor.

Reply:
If listener in AMQPClient read from rabbit, it' ll use callback function from MessagePublisher (onMessage), that has the same alias. Is it necessary to add the alias to messageProperty